src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java

```diff
@@ -0,0 +1,255 @@
+                             + "{*}" + submitUrl + "{*}" + submitUrl;
+        
+        verifyEmail(emails.get(0), student1.email, subject, textInEmail);
+        
+        verifyEmail(emails.get(1), student2.email, subject, "The following feedback session is now open");
+        
+        ______TS("feedback session reminders");
+        
+        emails = new EmailGenerator().generateFeedbackSessionReminderEmails(session, students, instructors, instructors);
+        assertEquals(15, emails.size());
+        
+        subject = String.format(EmailType.FEEDBACK_SESSION_REMINDER.getSubject(),
+                                course.getName(), session.getFeedbackSessionName());
+        
+        submitUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                          .withCourseId(course.getId())
+                          .withSessionName(session.getFeedbackSessionName())
+                          .toAbsoluteString();
+        
+        textInEmail = "Hello " + instructor1.name + "{*}The following feedback session is "
+                      + "still open for submissions{*}" + course.getId() + "{*}" + course.getName()
+                      + "{*}" + session.getFeedbackSessionName() + "{*}" + submitUrl + "{*}" + submitUrl;
+        
+        verifyEmail(emails.get(1), instructor1.email, subject, textInEmail);
+        
+        verifyEmail(emails.get(2), instructor2.email, subject,
+                    "The following feedback session is still open for submissions");
+        
+        ______TS("feedback session closing alerts");
+        
+        emails = new EmailGenerator().generateFeedbackSessionClosingEmails(session);
+        assertEquals(8, emails.size());
+        
+        subject = String.format(EmailType.FEEDBACK_CLOSING.getSubject(),
+                                course.getName(), session.getFeedbackSessionName());
+        
+        submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                          .withCourseId(course.getId())
+                          .withSessionName(session.getFeedbackSessionName())
+                          .withRegistrationKey(StringHelper.encrypt(student1.key))
+                          .withStudentEmail(student1.email)
+                          .toAbsoluteString();
+        
+        textInEmail = "Hello " + student1.name + "{*}The following feedback session is closing soon"
+                      + "{*}" + course.getId() + "{*}" + course.getName()
+                      + "{*}" + session.getFeedbackSessionName()
+                      + "{*}" + submitUrl + "{*}" + submitUrl
+                      + "{*}You may ignore this email if you have already submitted feedback.";
+        
+        verifyEmail(emails.get(0), student1.email, subject, textInEmail);
+        
+        verifyEmail(emails.get(1), student3.email, subject, "The following feedback session is closing soon");
```

Comment:
Just curious, is there any particular order `emails` is ordered by? `emails.get(1)` will always be the email for `student3` and not `student1`?